packages/lesswrong/components/notifications/NotificationsPage/NotificationsPageFeed.tsx

```diff
@@ -0,0 +1,231 @@
+import type { KarmaChanges } from "../../../lib/collections/users/karmaChangesGraphQL";
+import type { KarmaChangeUpdateFrequency } from "../../../lib/collections/users/schema";
+
+const styles = (theme: ThemeType) => ({
+  root: {
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    width: 760,
+    maxWidth: "100%",
+    margin: "0 auto",
+  },
+  tabs: {
+    marginBottom: 24,
+    "& .MuiTabs-flexContainer": {
+      gap: "32px",
+    },
+    "& .MuiTab-root": {
+      minWidth: 100,
+      [theme.breakpoints.down("xs")]: {
+        minWidth: 50,
+      },
+    },
+    "& .MuiTab-labelContainer": {
+      fontSize: 14,
+      fontWeight: 700,
+      letterSpacing: "0.28px",
+      textTransform: "uppercase",
+    },
+  },
+  karmaChanges: {
+    display: "flex",
+    flexDirection: "column",
+    gap: "10px",
+    marginTop: 10,
+  },
+  karmaBatching: {
+    marginBottom: 32,
+    fontSize: 14,
+    fontWeight: 500,
+    color: theme.palette.grey[600],
+    "& a": {
+      color: theme.palette.primary.dark,
+      fontWeight: 600,
+      marginLeft: 4,
+    },
+  },
+  notifications: {
+    display: "flex",
+    flexDirection: "column",
+    gap: "20px",
+    marginTop: 10,
+  },
+});
+
+// We have to do this manually outside of `usePaginatedResolver` because the
+// return type is pure unadulterated JSON, not a registered fragment type
+const query = gql`
+  query getNotificationDisplays($limit: Int, $type: String) {
+    NotificationDisplays(limit: $limit, type: $type) {
+      results
+    }
+  }
+`;
+
+const DEFAULT_LIMIT = 20;
+
+const batchingMessages: Record<KarmaChangeUpdateFrequency, string> = {
+  disabled: "Karma change batching is disabled",
+  daily: "Karma changes are batched daily",
+  weekly: "Karma changes are batched weekly",
+  realtime: "Karma changes are shown in realtime",
+};
+
+export const NotificationsPageFeed = ({
+  karmaChanges,
+  currentUser,
+  classes,
+}: {
+  karmaChanges?: KarmaChanges,
+  currentUser: UsersCurrent,
+  classes: ClassesType<typeof styles>,
+}) => {
+  const [limit, setLimit] = useState(DEFAULT_LIMIT);
+  const {tab, setTab} = useNotificationsPageTab();
+
+  const {
+    data,
+    error,
+    loading,
+    networkStatus,
+  } = useQuery(query, {
+    ssr: true,
+    notifyOnNetworkStatusChange: true,
+    skip: !currentUser,
+    pollInterval: 0,
+    fetchPolicy: "cache-and-network",
+    nextFetchPolicy: "cache-only",
+    variables: {
+      type: tab.type,
+      limit,
+    },
+  });
+
+  if (error) {
+    // eslint-disable-next-line no-console
+    console.error("Error loading notifications:", error);
+  }
+
+  const loadMore = useCallback(() => {
+    setLimit((currentLimit) => currentLimit + DEFAULT_LIMIT);
+  }, []);
+
+  const notifications = useRef<NotificationDisplay[]>([]);
+  if (data?.NotificationDisplays?.results && !loading) {
+    notifications.current = data.NotificationDisplays.results;
+  }
+
+  const canLoadMore = notifications.current.length > 0;
```

Comment:
Currently "Load more" is always shown when the tab has notifications right? I assume this is something you intended to return and finish later? But I would suggest making an asana ticket and fixing it in a future PR.